partner_tag_smart_assignation/models/res_partner_category_extension.py

```diff
@@ -0,0 +1,22 @@
+from odoo import fields, models, api
+
+
+class ResPartnerCategoryExtension(models.Model):
+    _inherit = "res.partner.category"
+
+    author_of_the_tag = fields.Many2one('res.users', string="Author")
+    department_that_uses_the_tag = fields.Many2one('hr.department', string="Department")
+    description_of_the_tag = fields.Text(string="Description")
```

Comment:
There are field naming conventions based upon their type. Especially for relational field, they should be suffixed by `_id` or `_ids`. I propose this:
```suggestion
    user_id = fields.Many2one('res.users', "Author", domain=[("share", "=", False)])  # domain is here to avoid putting external users
    department_id = fields.Many2one('hr.department', "Department")
    description = fields.Char()  # We don't want a large description, char field should be enough.
```